internal/build/systemd_sysext.go

```diff
@@ -46,19 +48,36 @@ func (b *Builder) downloadSystemExtensions(ctx context.Context, def *image.Defin
 	}
 
 	for _, extension := range extensions {
-		b.System.Logger().Info("Downloading extension %s from %s...",
+		b.System.Logger().Info("Pulling extension %s from %s...",
 			extension.Name, extension.Image)
 
-		extensionPath := filepath.Join(extensionsDir, filepath.Base(extension.Image))
+		if isRemoteURL(extension.Image) {
+			extensionPath := filepath.Join(extensionsDir, filepath.Base(extension.Image))
+			if err = b.DownloadFile(ctx, fs, extension.Image, extensionPath); err != nil {
+				return fmt.Errorf("downloading systemd extension %s: %w", extension.Name, err)
+			}
 
-		if err = b.DownloadFile(ctx, fs, extension.Image, extensionPath); err != nil {
-			return fmt.Errorf("downloading systemd extension %s: %w", extension.Name, err)
+			continue
+		}
+
+		unpacker := unpack.NewOCIUnpacker(b.System, extension.Image, unpack.WithLocalOCI(b.Local))
```

Comment:
I contemplated using a generic unpacker here but that would increase the wrapping by one level, e.g. if someone wants to provide an extension from a local directory, they'd have to create their extension as a directory or a disk first, and then embed that in another directory here. The confusion for disks would be even bigger in the sense that regardless of their origin (remote or local), we'd expect them to be ready to use as is and not attempt to unpack them.

Considering that this feature will firstly be used only for release manifests, I figured we can start out with OCI, and then revisit once we start thinking on the user (not consumer!) extensions story.